charts/nitro/templates/_helpers.tpl

```diff
@@ -149,6 +149,44 @@ Currently primarily used for stateless validator configuration
 {{- /* Return the processed JSON data */ -}}
 {{- $processed -}}
 
+{{- if and .Values.validator .Values.validator.splitvalidator .Values.validator.splitvalidator.global .Values.validator.splitvalidator.global.resources .Values.validator.splitvalidator.global.resources.limits .Values.validator.splitvalidator.global.resources.limits.memory -}}
+{{- $memory := .Values.validator.splitvalidator.global.resources.limits.memory -}}
+{{- $value := regexFind "^\\d*\\.?\\d+" $memory | float64 -}}
+{{- $unit := regexFind "[A-Za-z]+" $memory -}}
+{{- $valueMi := 0.0 -}}
+{{- if eq $unit "Gi" -}}
+  {{- $valueMi = mulf $value 1024 -}}
+{{- else if eq $unit "Mi" -}}
+  {{- $valueMi = $value -}}
+{{- end }}
+{{- if $.Values.env.goMemLimit.enabled }}
+- name: GOMEMLIMIT
+  value: {{ printf "%dMiB" (int (mulf $valueMi ($.Values.env.goMemLimit.multiplier | default 0.75))) }}
+{{- end }}
+{{- end -}}
+
```

Comment:
This is being added to the configprocessor named template. It should be in the `{{- define "nitro.splitvalidator.env" -}}` named template which is declared earlier in the file.